deep_qa/data/datasets/language_modelling/language_modelling_dataset.py

```diff
@@ -0,0 +1,32 @@
+from typing import List
+
+from overrides import overrides
+
+from ..dataset import TextDataset, count_labels
+from ...instances import TextInstance
+from ...instances.language_modeling import SentenceInstance
+from ....common.params import Params
+
+
+class LanguageModellingDataset(TextDataset):
+
+    def __init__(self, instances: List[TextInstance], params: Params=None):
+        self.sequence_length = params.pop("sequence_length")
+        super(LanguageModellingDataset, self).__init__(instances)
+
+    @staticmethod
+    @overrides
+    def read_from_file(filename: str, instance_class, params: Params=None):
+
+        sequence_length = params.get("sequence_length", 20)
+        with open(filename, "r") as text_file:
+            text = text_file.readlines()
+            text = " ".join([x.replace("\n", " ").strip() for x in text]).split(" ")
+
+        instances = []
+        for index in range(0, len(text) - sequence_length, sequence_length):
+            word_sequence = " ".join(text[index: index + sequence_length])
```

Comment:
Yeah - I was going to replace the `sentence_length` attribute of the language modelling dataset to be `approximate_sentence_length` for this very reason. I'll add a note. I think if we could get the tokeniser etc passed as an argument to Dataset and then the `Instances` using that tokeniser, instead of setting the class attribute as we currently do it, would be better, but that's a separate discussion.